examples/rosa-classic-public-with-shared-vpc/main.tf

```diff
@@ -0,0 +1,119 @@
+provider "aws" {
+  alias = "shared-vpc"
+
+  access_key = var.shared_vpc_aws_key
+  secret_key = var.shared_vpc_aws_secret
+  region     = var.shared_vpc_aws_region
+}
+
+############################
+# VPC
+############################
+module "vpc" {
+  source = "../../modules/vpc"
+
+  providers = {
+    aws = aws.shared-vpc
+  }
+
+  name_prefix  = var.cluster_name
+  subnet_count = var.replicas
+}
+
+locals {
+  account_role_prefix  = var.account_role_prefix != null ? var.account_role_prefix : "${var.cluster_name}-account"
+  shared_vpc_role_name = var.shared_vpc_role_name != null ? var.shared_vpc_role_name : "${var.cluster_name}-shared-vpc-role"
+  operator_role_prefix = var.operator_role_prefix != null ? var.operator_role_prefix : "${var.cluster_name}-operator"
+}
+
+##############################################################
+# Account roles includes IAM roles and IAM policies
+##############################################################
+module "account_iam_resources" {
+  source = "../../modules/account-iam-resources"
+
+  account_role_prefix = local.account_role_prefix
+  openshift_version   = var.openshift_version
+}
+
+############################
+# operator policies
+############################
+module "operator_policies" {
+  source = "../../modules/operator-policies"
+
+  account_role_prefix = module.account_iam_resources.account_role_prefix
+  openshift_version   = module.account_iam_resources.openshift_version
+  shared_vpc_role_arn = "arn:aws:iam::${var.shared_vpc_aws_account}:role/${local.shared_vpc_role_name}"
+}
+
+############################
+# OIDC provider
+############################
+module "oidc_provider" {
+  source = "../../modules/oidc-provider"
+
+  managed = true
+}
+
+############################
+# operator roles
+############################
+module "operator_roles" {
+  source = "../../modules/operator-roles"
+  # adding a dependency to operator_policies module
```

Comment:
why? if someone will remove it what he suppose to see/get?

Reply:
Its actually not needed, will remove it

Reply:
I removed this dependency and added time_wait resources to the relevant modules